File: 30_days_C#_HackerRank_Challenge/Scope.cs
public Difference(int[] elements)
    {
        this.elements = elements;
    }

    public void computeDifference()
    {
        int min = int.MaxValue;
        int max = int.MinValue;

        foreach (int x in elements)
        {
            if (x < min) min = x;
            if (x > max) max = x;
        }

        maximumDifference = max - min;
    }
}

File: 30_days_Python_HackerRank_Challenge/Scope.py
#python-2
class Difference:
    def __init__(self, a):
        self.__elements = a

        self.__elements = a
        self.maximumDifference = 0

    def computeDifference(self):
        self.maximumDifference = max(self.__elements) - min(self.__elements)

# End of Difference class

_ = raw_input()
a = [int(e) for e in raw_input().split(' ')]

d = Difference(a)
d.computeDifference()

print d.maximumDifference

#Python-3
class Difference:
    def __init__(self, a):
        self.__elements = a

        self.maximumDifference = 0

    def computeDifference(self):
        self.maximumDifference = max(self.__elements) - min(self.__elements)

# End of Difference class

_ = input()
a = [int(e) for e in input().split(' ')]

d = Difference(a)
d.computeDifference()

print(d.maximumDifference)
